src/generator/models.ts

```diff
@@ -158,10 +158,12 @@ class StructDef {
         readOnly = ` azure:"ro"`;
       }
       let tag = ` \`${this.Language.marshallingFormat}:"${serialization}"${readOnly}\``;
-      // if this is a response type then omit the tag IFF the marshalling format is
-      // JSON, it's a header or is the RawResponse field.  XML marshalling needs a tag.
+      // if this is a response type then omit the tag IFF the marshalling formatvalue is non-scalar
+      // for JSON marshalling types or it's a header.  XML marshalling needs a tag.
       // also omit the tag for additionalProperties
-      if ((this.Language.responseType === true && (this.Language.marshallingFormat !== 'xml' || prop.language.go!.name === 'RawResponse')) || prop.language.go!.isAdditionalProperties) {
+      if (prop.language.go!.name === 'RawResponse' && prop.language.go!.addTag === true && this.Language.marshallingFormat !== 'xml') {
+        tag = '`json:"-"`';
+      } else if (prop.language.go!.name === 'RawResponse' || prop.language.go!.isAdditionalProperties || (this.Language.responseType === true && prop.language.go!.addTag !== true && this.Language.marshallingFormat !== 'xml')) {
```

Comment:
This can also be simplified to remove the need for `addTag`, just call `isScalar()` here.  Note you might need to make this its own `if` clause (not an `else if`) given that the previous `if` clause is compound.

Reply:
So the reason why addTag is also necessary here is almost the same as above, with the addition that addTag is only added on scalar types being returned in the body and not the headers. There are response envelopes that return scalar types in headers and those are added on the response envelopes. So calling isScalar wont be sufficient for ruling out the header case.